app/controllers/customers_controller.rb

```diff
@@ -0,0 +1,95 @@
+class CustomersController < ApplicationController
+  before_action :set_customer, only: [:edit, :update, :destroy]
+
+  # GET /customers
+  # GET /customers.json
+  def index
+    @customers = Customer.all
+  end
+
+  # GET /customers/1
+  # GET /customers/1.json
+  def show
+    if Customer.exists?(params[:id])
+      @customer = Customer.find(params[:id])
+    else
+      redirect_to customers_url
+    end
+  end
+
+  # GET /customers/new
+  def new
+    @customer = Customer.new
+  end
+
+  # GET /customers/1/edit
+  def edit
+  end
+
+  # POST /customers
+  # POST /customers.json
```

Comment:
looks good, usually, it's best practices for create method to follow after the index, makes controller easier to read and track.